### DISCUSSION TITLE
Update request

### DISCUSSION THREAD

**Discussion by qomarhsn** [OpenBangla/OpenBangla-Keyboard#354] [General]
When can we expect the next update for the Open Bangla keyboard? It has been a while since the last update, and there are some special features that need to be implemented, which could elevate the Open Bengali keyboard to a higher position. Additionally, if some improvements are made to the phonetic key mapping, it would enhance the typing experience. Avro has also stopped updating, so conveying these concerns to them won't be of any assistance.

**Comment by mominul** [comment]
> if some improvements are made to the phonetic key mapping

@kamrulhasanio what improvements are you referring to, can you elaborate please?

**Reply by qomarhsn** [reply]
ফোনেটিকে যে সমস্ত বিশয় গুলো ইম্প্রুভ হলে ভালো হতো, 
* শব্দের মাঝে আ লেখতে হলে ` (accent) কী চেপে এরপর লেখতে হয় কেননা শব্দের মাঝে a শব্দকে য়া এর জন্য ম্যাপিং করে রাখা হয়েছে। অথচ য়া টা ya দিলেই লেখা যাচ্ছে। আর ফোনেটিক এর লেখার ধরণ অনুযায়ী এটাই অধিক যুক্তি যুক্ত। 
* ঁ টা ^ চিহ্ন এর কী থেকে সরিয়ে কোন একটা অক্ষরের কী এর সাথে দেয়া যেতে পারে, অক্ষর অক্ষরের কী'তে থাকাই অধিক যুক্তি যুক্ত হবার কথা অক্ষর চিহ্নের কী'তে থাকার চেয়ে। যেমন, A (shift+a). 
* ঃ টা : (কোলন) এর থেকে সরিয়ে অন্য কোন  অক্ষরের সাথে দেয়া উচিত। কেননা বর্তমানে ঃ থেকে : এর ব্যবহার বেশী। আর সরাসরি : এর স্থানে ঃ দেয়াই অনেকেই : এর স্থানে ঃ ব্যবহারে অভ্যস্ত হচ্ছে। তাই ঃ কে hh অথবা বর্তমান ঃ ও : এর ম্যাপিংটাই উলটো করে ম্যাপিং করা যেতে পারে।  
* হসন্ত টাও কোন একটা অক্ষরের সাথে আসা উচিত। কোন চিহ্নের সাথে দিলে সেটা ব্যবহারে বাধা সৃষ্টি হয়।অক্ষর হিসাবে G (sift+g) উত্তম চয়েজ হতে পারে। 
* নাম্বার পেডের . (ডট) চিহ্নটা . হিসাবেই রাখা উচিত এটাও দাড়ি হয়ে যাচ্ছে। অথচ এটাও বাংলায় একটা চিহ্ন হিসাবে ব্যবহার হয়। এখন দুটো .এ দাড়ি হয়ে যাওয়ায় এটা ব্যবহার করা যাচ্ছে না। (ল্যায়াউট ভিয়ার এ এটিকে ডট হিসাবে চিহ্নিত করলেও এটি দাড়িই ইনপুট নিচ্ছে।)

**Reply by qomarhsn** [reply]
And the feature that I wanted to mention as important has been addressed by a brother in [this post](https://github.com/OpenBangla/OpenBangla-Keyboard/discussions/372). Another crucial feature to add is custom keymapping. For this, there will be an option in the settings where users can map Bengali words as output for certain English letters. For example, one can set up ওপেন বাংলা কীবোর্ড mapping for the abbreviation "obk." When a user types "obk," it will automatically output ওপেন বাংলা কীবোর্ড.

**Reply by ghost** [reply]
I think the custom keymapping feature you are referring to can just be autocorrect entries,
For example, when a person types Vat, the suggestions window could list ভাত and "Value Added Text" together.
and for the obk example, you could create an auto-correct entry for "obk" that expands to ওপেন বাংলা কীবোর্ড.

I am not involved in development of OBK nor know of the inner workings of how OBK's auto-correct entry handling works, but if corrections that span multiple words and multiple corrections for a single word word can be implemented, that would solve this particular issue.

**Reply by qomarhsn** [reply]
The auto correction system will only be useful for the words added to the suggestion. And if custom key mapping can be added then each user can map specific word key for himself. Which may be a good thing depending on the person. Having said obk as an example, if someone needs many other words with it, he can add it.

**Reply by ghost** [reply]
> The auto correction system will only be useful for the words added to the suggestion. And if custom key mapping can be added then each user can map specific word key for himself. Which may be a good thing depending on the person. Having said obk as an example, if someone needs many other words with it, he can add it.

Auto-correct functions in almost the same exact way that you have mentioned, but it could be improvised.
